### PR TITLE
Make analysisd compile the CDB list upon start

### PR DESCRIPTION
**Pull request by carlocas**
|Related issue|
|--------|
|#3026|

## Description
ossec-makelist will be deprecated in future versions, so instead, ossec-analysisd will re-build the CBD list. That way, restarting the manager will apply the changes 

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation
- [x] Source upgrade
- Memory tests
   - [x] Valgrind report for affected components (`ossec-analysisd`) 

```
==20435== HEAP SUMMARY:
==20435==     in use at exit: 12,421 bytes in 305 blocks
==20435==   total heap usage: 12,696 allocs, 12,391 frees, 4,457,873 bytes allocated
==20434== LEAK SUMMARY:
==20434==    definitely lost: 100 bytes in 9 blocks
==20434==    indirectly lost: 0 bytes in 0 blocks
==20434==      possibly lost: 0 bytes in 0 blocks
==20434==    still reachable: 12,321 bytes in 296 blocks
==20434==         suppressed: 0 bytes in 0 blocks
```
   - [x] No errors found with scan-build
   - [x] No errors found with address-sanitizer.

- Custom tests
   - [x] Create a CBD list to check if ossec-analysisd compile the list
   - [x] Create a rule to check if the list is being read  
        create a CBD list with the IP 1.1.1.1, launch the event `Dec 10 01:02:02 host sshd[1234]: Failed none for root from 1.1.1.1 port 1066 ssh2` restart the manager and the event match the rule
```
**Phase 1: Completed pre-decoding.
       full event: 'Dec 10 01:02:02 host sshd[1234]: Failed none for root from 1.1.1.1 port 1066 ssh2'
       timestamp: 'Dec 10 01:02:02'
       hostname: 'host'
       program_name: 'sshd'
       log: 'Failed none for root from 1.1.1.1 port 1066 ssh2'

**Phase 2: Completed decoding.
       decoder: 'sshd'
       dstuser: 'root'
       srcip: '1.1.1.1'
       srcport: '1066'

**Phase 3: Completed filtering (rules).
       Rule id: '100001'
       Level: '5'
       Description: 'sshd: authentication failed from IP 1.1.1.1.'
```
The rule match the CBD list, launch the event `Dec 10 01:02:02 host sshd[1234]: Failed none for root from 3.3.3.3 port 1066 ssh2`
```
**Phase 1: Completed pre-decoding.
       full event: 'Dec 10 01:02:02 host sshd[1234]: Failed none for root from 3.3.3.3 from port 1066 ssh2'
       timestamp: 'Dec 10 01:02:02'
       hostname: 'host'
       program_name: 'sshd'
       log: 'Failed none for root from 3.3.3.3 from port 1066 ssh2'

**Phase 2: Completed decoding.
       decoder: 'sshd'

**Phase 3: Completed filtering (rules).
       Rule id: '5716'
       Level: '5'
       Description: 'sshd: authentication failed.'
```
The rule does not match the list, add the IP 2.2.2.2 to the list and launch the event again.

### CODE REVIEW

**Comment by chemamartinez** [comment]
We have to consider to change the name of the binary to build the CDB lists, from `ossec-makelists` to other more descriptive such as `compile-cdblists`. Ping @snaow, @carlocas, @wazuh/core.

File: src/analysisd/analysisd.c [hunk 1]
```diff
@@ -547,6 +550,7 @@ int main_analysisd(int argc, char **argv)
                 free(Config.lists);
                 Config.lists = NULL;
             }
+	    Lists_OP_MakeAll(force, show_message);
```
Comment (JuantAldea) [inline src/analysisd/analysisd.c]: Correct Indentation.
Variable force & show message aren't necessary.

File: src/analysisd/lists_make.c [hunk 2]
```diff
@@ -150,6 +152,8 @@ void Lists_OP_MakeCDB(const char *txt_filename, const char *cdb_filename, int fo
             return;
         }
     } else {
-        printf(" * File %s does not need to be compiled\n", cdb_filename);
+        if(show_message){
```
Comment (JuantAldea) [inline src/analysisd/lists_make.c]: Use `} else if () {` structure here.

File: src/analysisd/makelists.c [hunk 3]
```diff
@@ -64,6 +64,7 @@ int main(int argc, char **argv)
     uid_t uid;
     gid_t gid;
     int force = 0;
+    int show_message=1;
```
Comment (JuantAldea) [inline src/analysisd/makelists.c]: Unnecessary variable.

File: src/analysisd/makelists.c [hunk 4]
```diff
@@ -173,7 +174,7 @@ int main(int argc, char **argv)
         Config.lists = NULL;
     }
 
-    Lists_OP_MakeAll(force);
-
+    Lists_OP_MakeAll(force, show_message);
+    printf(" * The CBD list will be compiled automatically at restart of Wazuh\n");
```
Comment (JuantAldea) [inline src/analysisd/makelists.c]: Please change log message to
printf(" * The CBD list will be compiled automatically once Wazuh is restarted.\n");

File: src/analysisd/analysisd.c [hunk 5]
```diff
@@ -547,6 +548,7 @@ int main_analysisd(int argc, char **argv)
                 free(Config.lists);
                 Config.lists = NULL;
             }
+	        Lists_OP_MakeAll(0, 0);
```
Comment (JuantAldea) [inline src/analysisd/analysisd.c]: Correct Indentation

**Comment by chemamartinez** [comment]
Hi @carlocas, please check that when running `ossec-logtest`, the CDB lists are built as well.

**Comment by branchnetconsulting** [comment]
> Hi @carlocas, please check that when running `ossec-logtest`, the CDB lists are built as well.

I agree with Chema.  If CDBs get rebuilt also when ossec-logtest starts, then there would be no need for users to know about ossec-makelists or perhaps even to maintain a standalone binary for that purpose at all.  I love it when things get simpler :)

**Comment by carlocas** [comment]
I've been testing if `ossec-logtest` compiled the CBD list or not. In this test, I check that the CBD list didn't compile. In consequence, I've been working on a solution to this problem and the last commit resolved the issue. Now, with `ossec-logtest`, the CBD list can be compiled.

**Comment by jctello** [comment]
Great, I see you've also removed the BuildCDB function from the ossec-control utility accordingly.

File: src/analysisd/lists_make.c [hunk 6]
```diff
@@ -47,7 +47,9 @@ void Lists_OP_MakeCDB(const char *txt_filename, const char *cdb_filename, int fo
 
     if (File_DateofChange(txt_filename) > File_DateofChange(cdb_filename) ||
             force) {
-        printf(" * File %s needs to be updated\n", cdb_filename);
+    	if (show_message){
+            printf(" * File %s has been updated\n", cdb_filename);
```
Comment (chemamartinez) [inline src/analysisd/lists_make.c]: Much better: `CDB list %s has been updated successfully`

File: src/analysisd/lists_make.c [hunk 7]
```diff
@@ -153,7 +155,7 @@ void Lists_OP_MakeCDB(const char *txt_filename, const char *cdb_filename, int fo
             merror("Could not chmod cdb list '%s' to 660 due to: [%d - %s]", cdb_filename, errno, strerror(errno));
             return;
         }
-    } else {
+    } else if(show_message){
         printf(" * File %s does not need to be compiled\n", cdb_filename);
```
Comment (chemamartinez) [inline src/analysisd/lists_make.c]: The same here: `CDB list %s is up-to-date`

File: src/analysisd/makelists.c [hunk 8]
```diff
@@ -173,7 +173,8 @@ int main(int argc, char **argv)
         Config.lists = NULL;
     }
 
-    Lists_OP_MakeAll(force);
-
+    printf(" * This module has been deprecated.\n");
```
Comment (chemamartinez) [inline src/analysisd/makelists.c]: Specify the Wazuh version: `Since Wazuh v3.11.0, this module is deprecated`

File: src/analysisd/makelists.c [hunk 9]
```diff
@@ -173,7 +173,8 @@ int main(int argc, char **argv)
...
+    printf(" * The CBD list will be compiled automatically once Wazuh is restarted.\n");
```
Comment (chemamartinez) [inline src/analysisd/makelists.c]: `CDB lists are now compiled at manager start-up time automatically`

**Comment by branchnetconsulting** [comment]
To "CDB lists are now compiled at manager start-up time automatically"
you might want to add
"as well as each time ossec-logtest is run."

**Pull request merged**